contributors/design-proposals/key-set-as-merge-key.md

```diff
@@ -0,0 +1,336 @@
+# Key Set as Merge Key in Strategic Merge Patch
+
+## Abstract
+
+Support a key set as the merge key in Strategic Merge Patch.
+
+## Motivation
+
+*Background*: In Strategic Merge Patch, Merge Key is the key to distinguish the entries in the list of non-primitive types.
+It must always be present and unique to perform the merge on the list of non-primitive types,
+and will be preserved.
+
+The issue in practice is that sometimes we cannot find a key that always be present and unique.
```

Comment:
I'd replace this with:

>
The current implementation requires a single field that uniquely identifies each element in a list.  For some element Kinds, the identity is defined using multiple fields.  An example of this is the `x.x.xPort`, which is identified by `protocol` and `port`